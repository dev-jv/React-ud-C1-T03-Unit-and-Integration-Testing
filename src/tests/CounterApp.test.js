import React from "react";
import '@testing-library/jest-dom/extend-expect'
import CounterApp from "../CounterApp";
import { shallow } from 'enzyme';
import '../setupTests'
import colors from 'colors';

describe('/> Testing in <CounterApp />'.bgBlue.white, () => {

    let wrapper = shallow(<CounterApp />); // Declared for all tests, wrapper can be modified as tests progress
    // It is initialized so as not to lose the intellisense
    beforeEach( () => {
        wrapper = shallow(<CounterApp />); // The same wrapper for each test
    });

    test('Test 1: Should show <CounterApp /> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Test 2: Should show the past value', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value={ value }/>);
        const num = wrapper.find('h2').text().trim();
        expect(parseInt(num)).toBe(100);
    });

    test('Test 3: Should increment with the +1 button', () => {
        wrapper.find('button').at(0).simulate('click'); // simulate(click, {...});
        const txt = wrapper.find('h2').text().trim();
        expect(txt).toBe('1')
    });

    test('Test 4: Should decrement with the -1 button', () => {
        wrapper.find('button').at(2).simulate('click');
        const txt = wrapper.find('h2').text().trim();
        // console.log(txt);
        expect(txt).toBe('-1');
    });

    test('Test 5: Should set the default value with the reset btn', () => {
        const value = 0;
        const wrapper = shallow(<CounterApp value={ value }/>);

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const res = wrapper.find('h2').text().trim();
        // console.log(res);
        expect(res).toBe('0')
    })

});
