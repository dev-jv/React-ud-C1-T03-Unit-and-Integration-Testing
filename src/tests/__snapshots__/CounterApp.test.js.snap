// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[37m[44m/> Testing in <CounterApp />[49m[39m Test 1: Should show <CounterApp /> correctly 1`] = `
<Fragment>
  <h1>
     CounterApp 
  </h1>
  <h2>
    0
  </h2>
  <button
    id="hAdd"
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    id="hReset"
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    id="hSubtract"
    onClick={[Function]}
  >
     -1 
  </button>
</Fragment>
`;
